src/chain/chain.rs

```diff
@@ -1457,7 +1457,7 @@ impl Chain {
                 }
                 self.candidates(target_name)?
             }
-            Location::PrefixSection(ref prefix) => {
+            DstLocation::Prefix(ref prefix) => {
```

Comment:
I know you wanted to remove it but I would prefer to do it in a separate PR because it's more involved - required changes in the `NeighbourInfo` messages flow.